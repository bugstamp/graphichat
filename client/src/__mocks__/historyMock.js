export const historyPushMock = jest.fn(str => str);

const historyMock = {
  push: historyPushMock,
};

export default historyMock;
