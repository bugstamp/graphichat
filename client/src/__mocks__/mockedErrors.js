import { createError } from 'apollo-errors';

export const AuthenticationError = createError('AuthenticationError', {
  message: 'The user is not authorized',
});

export const BadInputError = createError('BadInputError', {
  message: 'The provided value is invalid',
});

export const ForbiddenError = createError('ForbiddenError', {
  message: 'Access is denied',
});

export const ServerError = createError('ServerError', {
  message: 'Server Error',
});

export class GraphQLErrors extends Error {
  constructor(graphQLErrors) {
    super();
    this.graphQLErrors = graphQLErrors;
    this.name = 'GraphQLErrors';
  }
}
