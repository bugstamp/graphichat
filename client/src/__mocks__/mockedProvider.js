import React from 'react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { InMemoryCache } from 'apollo-boost';
import { MockedProvider } from '@apollo/react-testing';
import { MuiThemeProvider } from '@material-ui/core/styles';
import { ThemeProvider } from 'styled-components';
import { mount } from 'enzyme';

import { initialState } from '../apollo/cache';
import { theme } from '../styles';

// eslint-disable-next-line
const TestProvider = ({ children, mocks, cache, history }) => (
  <MuiThemeProvider theme={theme}>
    <ThemeProvider theme={theme}>
      <MockedProvider
        mocks={mocks}
        cache={cache}
        addTypename={false}
      >
        <Router history={history}>
          {children}
        </Router>
      </MockedProvider>
    </ThemeProvider>
  </MuiThemeProvider>
);

export const mountMockedProvider = (children, mocks = [], {
  history = createMemoryHistory(),
  state = null,
} = {}) => {
  const cache = new InMemoryCache({ addTypename: false });
  cache.writeData({ data: state || initialState });

  return mount((
    <TestProvider
      mocks={mocks}
      cache={cache}
      history={history}
    >
      {children}
    </TestProvider>
  ));
};

export default TestProvider;
