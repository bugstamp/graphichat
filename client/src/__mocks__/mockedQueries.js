import gql from '../gql';
import {
  avatar,
  me,
  user,
  userActivityUpdate,
  contact,
  chat,
} from './mockedQueryData';

const {
  CHECK_SESSION_EXPIRATION,
  GET_INITIAL_DATA,
  UPDATE_USER,
  UPLOAD_AVATAR,
  USER_UPDATE_SUBSCRIPTION,
  USER_ACTIVITY_SUBSCRIPTION,
  CHAT_CREATED_SUBSCRIPTION,
  SIGN_OUT,
} = gql;

export const checkSessionExpirationMock = {
  request: {
    query: CHECK_SESSION_EXPIRATION,
    variables: {},
  },
  result: {
    data: {
      me,
      myContacts: [],
      myChats: [],
    },
  },
};
export const getInitialDataMock = {
  request: {
    query: GET_INITIAL_DATA,
    variables: {},
  },
  result: {
    data: {
      me,
      myContacts: [],
      myChats: [],
    },
  },
};
export const updateUserMock = {
  request: {
    query: UPDATE_USER,
    variables: {
      field: 'username',
      value: 'user',
    },
  },
  result: {
    data: {
      updateUser: me,
    },
  },
};
export const uploadAvatarMock = {
  request: {
    query: UPLOAD_AVATAR,
    variables: {
      file: {},
    },
  },
  result: {
    data: {
      uploadAvatar: avatar,
    },
  },
};
export const userUpdateSubscriptionMock = {
  request: {
    query: USER_UPDATE_SUBSCRIPTION,
    variables: {},
  },
  result: {
    data: {
      userUpdated: me,
    },
  },
};
export const userActivitySubscriptionMock = {
  request: {
    query: USER_ACTIVITY_SUBSCRIPTION,
    variables: {},
  },
  result: {
    data: {
      userActivityUpdated: userActivityUpdate,
    },
  },
};
export const chatCreatedSubscriptionMock = {
  request: {
    query: CHAT_CREATED_SUBSCRIPTION,
    variables: {},
  },
  result: {
    data: {
      chatCreated: {
        contact,
        chat,
      },
    },
  },
};
export const signOutMock = {
  request: {
    query: SIGN_OUT,
    variables: {
      userId: me.id,
    },
  },
  result: {
    data: true,
  },
};
