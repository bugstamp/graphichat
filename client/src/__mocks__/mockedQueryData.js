export const avatar = {
  sm: '',
  md: '',
  __typename: 'UserAvatar',
};

export const uploadAvatar = {
  ...avatar,
  sm: 'smAvatar',
  md: 'mdAvatar',
};

export const me = {
  id: 1,
  avatar,
  username: 'me',
  displayName: 'Me Test',
  firstName: 'Me',
  lastName: 'Test',
  status: 'ONLINE',
  lastDate: Date.now(),
  __typename: 'User',
};

export const user = {
  id: 2,
  avatar,
  username: 'user',
  displayName: 'User Test',
  firstName: 'User',
  lastName: 'Test',
  status: 'OFFLINE',
  lastDate: Date.now(),
  __typename: 'User',
};

export const userActivityUpdate = {
  userId: 2,
  status: 'ONLINE',
  lastDate: Date.now(),
};

export const contact = {
  id: 11,
  chatId: 1,
  userInfo: user,
};

export const message = {
  id: 1,
  senderId: 1,
  type: 'text',
  content: 'yo!',
  time: Date.now(),
  edited: false,
  seen: false,
};

export const chat = {
  id: 1,
  messages: [message],
};
