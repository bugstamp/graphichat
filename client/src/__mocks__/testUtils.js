// fix bug with react-facebook-login lib
export const fixReactFacebookLoginBug = () => {
  const script = document.createElement('script');
  script.type = 'text/javascript';
  document.getElementsByTagName('head')[0].appendChild(script);
};
