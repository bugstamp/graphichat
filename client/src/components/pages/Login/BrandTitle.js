import React from 'react';

import Typography from '@material-ui/core/Typography';
import useMediaQuery from '@material-ui/core/useMediaQuery';

import Logo from '../../common/Logo';
import { BrandTitleWrapper } from './styled';

const getLogoParams = (downMd, downSm) => {
  if (downSm) {
    return [35, 'h3'];
  }
  if (downMd) {
    return [35, 'h2'];
  }
  return [60, 'h1'];
};

const BrandTitle = () => {
  const downMd = useMediaQuery(theme => theme.breakpoints.down('md'));
  const downSm = useMediaQuery(theme => theme.breakpoints.down('sm'));
  const [logoSize, typographyVariant] = getLogoParams(downMd, downSm);

  return (
    <BrandTitleWrapper>
      <Logo size={logoSize} />
      <Typography
        component="p"
        variant={typographyVariant}
        align="center"
      >
        GraphiChat
      </Typography>
    </BrandTitleWrapper>
  );
};

export default BrandTitle;
