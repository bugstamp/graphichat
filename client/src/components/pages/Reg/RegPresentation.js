import React, { memo } from 'react';
import { map } from 'lodash';

import Typography from '@material-ui/core/Typography';

import BrandTitle from '../Login/BrandTitle';

import {
  RegPresentationWrapper,
  SubTitle,
  SubTitleWord,
  Enjoy,
} from './styled';
import { isEven } from '../../../helpers';

export const words = ['Simple.', 'Fun.', 'Fast.', 'Useful.', 'Powerful.'];

export const getWordPropsByIndex = (index) => {
  const num = index + 1;
  const order = isEven(num) ? 'even' : 'odd';
  const delay = (num / words.length) + 1;

  return { order, delay };
};

const RegPresentation = () => (
  <RegPresentationWrapper>
    <BrandTitle />
    <SubTitle>
      {
        map(words, (word, index) => {
          const wordProps = getWordPropsByIndex(index);

          return (
            <SubTitleWord
              key={word}
              component="p"
              variant="h4"
              align="center"
              {...wordProps}
            >
              {word}
            </SubTitleWord>
          );
        })
      }
    </SubTitle>
    <Enjoy>
      <Typography
        component="p"
        variant="h4"
        align="center"
        color="inherit"
      >
        Enjoy!
      </Typography>
    </Enjoy>
  </RegPresentationWrapper>
);

export { RegPresentation };

export default memo(RegPresentation);
