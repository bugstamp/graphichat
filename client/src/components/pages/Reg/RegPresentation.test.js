import React from 'react';
import { shallow } from 'enzyme';

import Typography from '@material-ui/core/Typography';

import RegPresentation, { words, getWordPropsByIndex } from './RegPresentation';
import BrandTitle from '../Login/BrandTitle';
import {
  SubTitle,
  SubTitleWord,
  Enjoy,
} from './styled';

import { mountMockedProvider } from '../../../__mocks__/mockedProvider';

describe('RegPresentation', () => {
  const mountWrapper = () => mountMockedProvider(<RegPresentation />);
  const wrapper = mountWrapper();

  test('should match to snapshot', () => {
    expect(shallow(<RegPresentation />)).toMatchSnapshot();
  });
  test('should mount without errors', () => {
    expect(wrapper.find(RegPresentation)).toHaveLength(1);
  });
  test('should have BrandTitle', () => {
    expect(wrapper.find(BrandTitle)).toHaveLength(1);
  });
  describe('SubTitle', () => {
    const SubTitleWords = wrapper.find(SubTitleWord);

    test('should be', () => {
      expect(wrapper.find(SubTitle)).toHaveLength(1);
    });
    test('should have correct amount of SubTitleWords', () => {
      expect(SubTitleWords).toHaveLength(words.length);
    });
    test('SubTitleWords text should be match to words array', () => {
      const texts = SubTitleWords.map(w => w.text());

      expect(texts).toEqual(expect.arrayContaining(words));
    });
    test('SubTitleWords static props', () => {
      SubTitleWords.forEach((w) => {
        expect(w.props()).toMatchObject({
          component: 'p',
          variant: 'h4',
          align: 'center',
        });
      });
    });
    test('SubTitleWords order and delay dynamic props', () => {
      SubTitleWords.forEach((w, index) => {
        const { order, delay } = getWordPropsByIndex(index);

        expect(w.prop('order')).toEqual(order);
        expect(w.prop('delay')).toEqual(delay);
      });
    });
  });
  describe('Enjoy', () => {
    const EnjoyWrapper = wrapper.find(Enjoy);

    test('should be', () => {
      expect(EnjoyWrapper).toHaveLength(1);
    });
    test('should have child paragraph with correct props', () => {
      expect(EnjoyWrapper.find(Typography)).toHaveLength(1);
      expect(EnjoyWrapper.find(Typography).props()).toMatchObject({
        component: 'p',
        variant: 'h4',
        align: 'center',
        color: 'inherit',
        children: 'Enjoy!',
      });
    });
  });
});
