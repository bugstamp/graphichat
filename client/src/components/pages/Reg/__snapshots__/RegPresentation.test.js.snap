// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegPresentation should match to snapshot 1`] = `
<styled__RegPresentationWrapper>
  <BrandTitle />
  <styled__SubTitle>
    <styled__SubTitleWord
      align="center"
      component="p"
      delay={1.2}
      key="Simple."
      order="odd"
      variant="h4"
    >
      Simple.
    </styled__SubTitleWord>
    <styled__SubTitleWord
      align="center"
      component="p"
      delay={1.4}
      key="Fun."
      order="even"
      variant="h4"
    >
      Fun.
    </styled__SubTitleWord>
    <styled__SubTitleWord
      align="center"
      component="p"
      delay={1.6}
      key="Fast."
      order="odd"
      variant="h4"
    >
      Fast.
    </styled__SubTitleWord>
    <styled__SubTitleWord
      align="center"
      component="p"
      delay={1.8}
      key="Useful."
      order="even"
      variant="h4"
    >
      Useful.
    </styled__SubTitleWord>
    <styled__SubTitleWord
      align="center"
      component="p"
      delay={2}
      key="Powerful."
      order="odd"
      variant="h4"
    >
      Powerful.
    </styled__SubTitleWord>
  </styled__SubTitle>
  <styled__Enjoy>
    <WithStyles(ForwardRef(Typography))
      align="center"
      color="inherit"
      component="p"
      variant="h4"
    >
      Enjoy!
    </WithStyles(ForwardRef(Typography))>
  </styled__Enjoy>
</styled__RegPresentationWrapper>
`;
